src/Products/urban/validators/validator.py

```diff
@@ -67,6 +67,34 @@ class isNotDuplicatedReferenceValidator:
     def __init__(self, name):
         self.name = name
 
+    def _check_similar_licences_ref(
+        self,
+        context_ref,
+        context_licence,
+        similar_licences
+    ):
+        if not similar_licences:
+            return True
+
+        output_check = []
+
+        for licence in similar_licences:
+            check_ref = (
+                licence.getObject().getReference().lower() == context_ref.lower()
+            )
+            check_uid = licence.UID == context_licence.UID()
```

Comment:
UID check should be done first to avoid over computation when this is not necessary